packages/wallet/src/redux/protocols/indirect-funding/reducer.ts

```diff
@@ -0,0 +1,149 @@
+import { SharedData, getExistingChannel } from '../../state';
+import { ProtocolStateWithSharedData } from '..';
+import { IndirectFundingState } from './states';
+import * as selectors from '../../selectors';
+import * as helpers from '../reducer-helpers';
+import { getLastCommitment } from '../../channel-store/channel-state';
+import { CommitmentType } from 'fmg-core';
+import {
+  initializeExistingLedgerFunding,
+  isExistingLedgerFundingAction,
+  existingLedgerFundingReducer,
+} from '../existing-ledger-funding';
+import * as states from './states';
+import {
+  initializeNewLedgerFunding,
+  isNewLedgerFundingAction,
+  newLedgerFundingReducer,
+} from '../new-ledger-funding';
+import { IndirectFundingAction } from './actions';
+export function initialize(
+  processId: string,
+  channelId: string,
+  sharedData: SharedData,
+): ProtocolStateWithSharedData<IndirectFundingState> {
+  const existingLedgerChannel = selectors.getExistingLedgerChannelForParticipants(
+    sharedData,
+    helpers.getOurAddress(channelId, sharedData),
+    helpers.getOpponentAddress(channelId, sharedData),
+  );
+
+  if (
+    existingLedgerChannel &&
+    getLastCommitment(existingLedgerChannel).commitmentType === CommitmentType.App
```

Comment:
Currently I don't think that will ever be the case since we always create update commitments after creating a ledger channel but I've changed the check to check for PostFund Setups so this will work in the future.